### ISSUE TITLE
Tiktok black screen for videos

### ISSUE THREAD

**Issue by billbeans** [waydroid/waydroid#1596]
### Describe the bug

For some reason, Tiktok seems to display a black screen in the place of videos, but only after restarting the app once. To get normal functionality again, you have to clear the cache and re-sign in. This is puzzling because the videos seem to play in the background, only you are met with a black screen.

### Waydroid version

1.4.3

### Device

Linux Desktop

### Operating System

Debian 12 Bookworm

### Kernel version

6.1.0-26-amd64

### Desktop Environment

GNOME 43.9

### GPU

AMD Radeon RX 6600

### Logs

[waydroid.cfg.txt](https://github.com/user-attachments/files/17416236/waydroid.cfg.txt)
[waydroid.prop.txt](https://github.com/user-attachments/files/17416238/waydroid.prop.txt)
[waydroid_base.prop.txt](https://github.com/user-attachments/files/17416240/waydroid_base.prop.txt)
[waydroid.log](https://github.com/user-attachments/files/17416243/waydroid.log)
[logcat.txt](https://github.com/user-attachments/files/17416247/logcat.txt)
[dmesg.txt](https://github.com/user-attachments/files/17416250/dmesg.txt)

**Comment by billbeans** [comment]
Some theories on why this is happening:

I suspect this is some kind of throttling done server-side to accounts they think are bots. There are many results on Google about this black screen issue and they all seem to be from people using Android phones.

I have been experimenting with Magisk and LSPosed modules to try and spoof a more convincing user agent, with no luck so far. When I tried Genymotion it seemed to work better with no black screen, but sometimes the videos still fail to load. Also, my accounts wouldn't be able to follow anyone, which I suspect is also because of the VM.

It would be really nice to have a functional Tiktok app without selling your soul to the devil by having it harvest all your personal information as it would if you installed it on your phone, for example. I would love to just have a personal account on my Linux desktop for browsing, the app is nice. Can a dev weigh in here, please?

**Comment by billbeans** [comment]
SOLVED

https://github.com/Quackdoc/waydroid-scripts/blob/main/spoof-device.sh

I still appear to be shadowbanned from following accounts, but this at least fixes the black screen problem
